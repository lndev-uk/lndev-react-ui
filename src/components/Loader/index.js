//@flow
import React from 'react';
import PropTypes from 'prop-types';

type LoaderType = 'bar' | 'spinner';

export type LoaderProps = {
  type: LoaderType,
  loading: Boolean
};
export default function Loader(props: LoaderProps): React.ReactNode {
  const { type, loading } = props;
  if (loading) {
    if (type === 'bar') {
      return <div className="loadingBar"></div>;
    } else {
      return (
        <div className="loadingSpinner">
          <span className="loadingSpinner-inner"></span>
          <span className="loadingSpinner-inner"></span>
          <span className="loadingSpinner-inner"></span>
          <span className="loadingSpinner-inner"></span>
        </div>
      );
    }
  }
  return null;
}

Loader.defaultProps = {
  loading: true
};

Loader.propTypes = {
  loading: PropTypes.bool,
  type: PropTypes.oneOf(['bar', 'spinner'])
};
