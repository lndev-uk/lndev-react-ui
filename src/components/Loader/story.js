import React from 'react';
import Loader from '.';

export default {
  title: 'Loader'
};

export const Bar = () => (
  <div>
    <Loader type="bar" loading={true} />
  </div>
);

export const Spinner = () => (
  <div>
    <div>
      <Loader type="spinner" loading={true} />
    </div>
  </div>
);
