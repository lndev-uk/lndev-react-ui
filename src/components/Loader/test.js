import React from 'react';
import Loader from '.';
import { render } from '@testing-library/react';

describe('Loader', () => {
  it('matches snapshot for bar', () => {
    const { container } = render(<Loader type="bar" />);
    expect(container).toMatchSnapshot('bar');
  });

  it('matches snapshot for spinner', () => {
    const { container } = render(<Loader type="spinner" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot loading = false', () => {
    const { container } = render(<Loader type="bar" loading={false} />);
    expect(container).toMatchSnapshot();
  });
  it('matches snapshot loading = true', () => {
    const { container } = render(<Loader type="bar" loading={true} />);
    expect(container).toMatchSnapshot();
  });
});
