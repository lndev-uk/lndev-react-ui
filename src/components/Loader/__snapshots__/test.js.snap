// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot for spinner 1`] = `
<div>
  <div
    class="loadingSpinner"
  >
    <span
      class="loadingSpinner-inner"
    />
    <span
      class="loadingSpinner-inner"
    />
    <span
      class="loadingSpinner-inner"
    />
    <span
      class="loadingSpinner-inner"
    />
  </div>
</div>
`;

exports[`Loader matches snapshot loading = false 1`] = `<div />`;

exports[`Loader matches snapshot loading = true 1`] = `
<div>
  <div
    class="loadingBar"
  />
</div>
`;

exports[`bar 1`] = `
<div>
  <div
    class="loadingBar"
  />
</div>
`;
